File: app/src/main/java/com/philsoft/metrotripper/prefs/Prefs.java
package com.philsoft.metrotripper.prefs;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.common.collect.Sets;

import org.apache.commons.lang.StringUtils;

import java.util.LinkedHashSet;
import java.util.Set;


public class Prefs {


	private static final String DEFAULT_PREFS_BUCKET = "defaultBucket";
	private static final String KEY_SAVED_STOPS = "KEY_SAVED_STOPS";
	private static final String KEY_LAST_STOP_ID = "KEY_LAST_STOP_ID";

	private static SharedPreferences getPrefs(Context ctx) {
		return ctx.getSharedPreferences(DEFAULT_PREFS_BUCKET, Context.MODE_PRIVATE);
	}

	public static void saveStopId(Context context, long stopId) {
		Set<Long> savedStops = getSavedStopIds(context);
		savedStops.add(stopId);
		String joinedStops = StringUtils.join(savedStops, ",");
		getPrefs(context).edit().putString(KEY_SAVED_STOPS, joinedStops).commit();
	}

	public static LinkedHashSet<Long> getSavedStopIds(Context context) {
		String joinedStops = getPrefs(context).getString(KEY_SAVED_STOPS, "");
		LinkedHashSet<Long> stopIds = Sets.newLinkedHashSet();
		for (String stopIdStr : StringUtils.split(joinedStops, ",")) {
			stopIds.add(Long.valueOf(stopIdStr));
		}
		return stopIds;
	}

	public static void removeStopId(Context context, long stopId) {
		Set<Long> savedStops = getSavedStopIds(context);
		savedStops.remove(stopId);
		String joinedStops = StringUtils.join(savedStops, ",");
		getPrefs(context).edit().putString(KEY_SAVED_STOPS, joinedStops).commit();
	}

	public static void saveLastStopId(Context context, long stopId) {
		getPrefs(context).edit().putLong(KEY_LAST_STOP_ID, stopId).commit();
	}

	public static long getLastStopId(Context context) {
		return getPrefs(context).getLong(KEY_LAST_STOP_ID, -1);
	}
}

File: app/src/main/java/com/philsoft/metrotripper/prefs/Prefs.kt
package com.philsoft.metrotripper.prefs

import android.content.Context
import org.apache.commons.lang.StringUtils
import java.util.*


class Prefs private constructor(context: Context) {
    private val prefs = context.getSharedPreferences(DEFAULT_PREFS_BUCKET, Context.MODE_PRIVATE)

    companion object {
        private const val DEFAULT_PREFS_BUCKET = "defaultBucket"
        private const val KEY_SAVED_STOPS = "KEY_SAVED_STOPS"

        private var instance: Prefs? = null
        @Synchronized
        @JvmStatic
        fun getInstance(ctx: Context): Prefs {
            if (instance == null) {
                instance = Prefs(ctx.applicationContext)
            }
            return instance!!
        }
    }

    private var savedStops: String
        get() = prefs.getString(KEY_SAVED_STOPS, "")
        set(stops) = prefs.edit().putString(KEY_SAVED_STOPS, stops).apply()

    fun saveStopId(stopId: Long) {
        val stops = getSavedStopIds()
        stops.add(stopId)
        savedStops = StringUtils.join(stops, ",")
    }

    fun getSavedStopIds(): LinkedHashSet<Long> {
        return savedStops.split(",").filterNot { it.isEmpty() }.mapTo(linkedSetOf()) { it.toLong() }
    }

    fun removeStopId(stopId: Long) {
        val stops = getSavedStopIds().filterNot { it == stopId }.joinToString(",")
        savedStops = stops
    }
}
